### ISSUE TITLE
binary heap retain issue

### ISSUE THREAD

**Issue by Pulecz** [dandels/dmodman#6]
during cargo build (or run) --release:
```
   Compiling reqwest v0.11.18
   Compiling dmodman v0.2.2 (/home/pulec/git/dmodman)
error[E0658]: use of unstable library feature 'binary_heap_retain'
  --> src/cache/mod.rs:94:14
   |
94 |         heap.retain(|fdata| fdata.file_id != id_to_delete);
   |              ^^^^^^
   |
   = note: see issue #71503 <https://github.com/rust-lang/rust/issues/71503> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `dmodman` due to previous error
```

I am not any smarter after scrolling through the talk at the [mentioned issue](https://github.com/rust-lang/rust/issues/71503).

rustc explain wants me to use rustc nightly or something?

Uff, rusty issue.

**Comment by dandels** [comment]
It happens that I used a very recent library feature here. BinaryHeap::retain was stabilized in [Rust 1.70](https://blog.rust-lang.org/2023/06/01/Rust-1.70.0.html#stabilized-apis) which was released June 1. Your version of Rust seems to be older than that, so it thinks it's still an unstable feature. While the readme states that the project works with the latest stable toolchain, I might've avoided using this if I realized it had just been released.
